### PR TITLE
[Merged by Bors] - feat: port Algebra.AddTorsor

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---
There are some dangerous instances here marked with `nolint dangerous_instance` in Lean3 that cause problems

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jsm28** [comment]
If it's the `add_torsor.nonempty` dangerous instance that causes problems, then it might be worth doing a preliminary refactor in mathlib3: as suggested in https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Empty.20torsors allow an `add_torsor` to be empty and instead add `nonempty` hypotheses at only those use sites that need them.

**Comment by ChrisHughes24** [comment]
> If it's the `add_torsor.nonempty` dangerous instance that causes problems, then it might be worth doing a preliminary refactor in mathlib3: as suggested in https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Empty.20torsors allow an `add_torsor` to be empty and instead add `nonempty` hypotheses at only those use sites that need them.

I think it's probably the `toVSub` one sadly.

File: Mathlib/Algebra/AddTorsor.lean [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+  right_inv v := by simp [vsub_vadd_eq_vsub_sub]
+#align equiv.const_vsub Equiv.constVsub
+
+@[simp]
+theorem coe_const_vsub (p : P) : ⇑(constVsub p) = (· -ᵥ ·) p :=
+  rfl
+#align equiv.coe_const_vsub Equiv.coe_const_vsub
+
+@[simp]
+theorem coe_const_vsub_symm (p : P) : ⇑(constVsub p).symm = fun v => -v +ᵥ p :=
```
Comment (ericrbg) [inline Mathlib/Algebra/AddTorsor.lean:398]:
```suggestion
theorem coe_const_vsub_symm (p : P) : (constVsub p).symm = (fun (v : G) => (-v +ᵥ p : P)) :=
```

putting all the type annotations seems to help this thing to elaborate, but I guess this isn't the big issue in this file

**Comment by LukasMias** [comment]
I just pushed some proofs which allow the instance to compile, though they are a lot uglier than what lean3 used to have. This is almost certainly sweeping a deeper problem under the rug, since the proofs feel unnatural and there is probably something messy going on with instance synthesization, but perhaps this can also help  someone to understand what the problem is.

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**